modules/apps/headless/headless-asset-library/headless-asset-library-impl/src/main/java/com/liferay/headless/asset/library/internal/util/comparator/AssetLibraryOrderByComparator.java

```diff
@@ -0,0 +1,124 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.headless.asset.library.internal.util.comparator;
+
+import com.liferay.portal.kernel.bean.BeanPropertiesUtil;
+import com.liferay.portal.kernel.search.Sort;
+import com.liferay.portal.kernel.util.CollatorUtil;
+import com.liferay.portal.kernel.util.LocalizationUtil;
+import com.liferay.portal.kernel.util.OrderByComparator;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.kernel.util.Validator;
+
+import java.text.Collator;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+
+/**
+ * @author Alicia García
+ */
+public class AssetLibraryOrderByComparator<T> extends OrderByComparator<T> {
+
+	public AssetLibraryOrderByComparator(Locale locale, Sort[] sorts) {
+		List<Object> columns = new ArrayList<>();
+
+		for (Sort sort : sorts) {
+			columns.add(sort.getFieldName());
+			columns.add(!sort.isReverse());
+		}
+
+		if (columns.isEmpty() || ((columns.size() % 2) != 0)) {
+			throw new IllegalArgumentException(
+				"Columns length is not an even number");
+		}
+
+		_locale = locale;
+
+		_columns = columns;
```

Comment:
Why are you creating the `_columns` list? You could simply store `sorts` as a field and iterate over it. I don't understand why this step is necessary :confused:

Reply:
we are using on the compare
`public int compare(T object1, T object2) {`

We can create this field after, but is necesary

Reply:
we are using on the compare
public int compare(T object1, T object2) {

We can create this field after, but is necesary

Reply:
I do not understand if this comments are sent or not...

Reply:
ok, 

this is sent?

ok , we can create a _sorts field, or change the _colums to be a map and be more clear, but I do not see the improvement, on making the first one. but I change it